client/finality-grandpa/src/communication/mod.rs

```diff
@@ -665,14 +667,24 @@ impl<Block: BlockT> Sink<Message<Block>> for OutgoingMessages<Block>
 		}
 
 		// when locals exist, sign messages on import
-		if let Some((ref pair, _)) = self.locals {
+		if let Some(ref public) = self.local_id {
+			let keystore = match &self.keystore {
+				Some(keystore) => keystore.clone(),
+				None => {
+					return Err(Error::Signing("Cannot sign without a keystore".to_string()))
```

Comment:
We won't need this if we have instead `Option<(AuthorityId, BareCryptoStorePtr)>`.